From e52c1a4476f6fe0ab4c66afa727d786853efe7ad Mon Sep 17 00:00:00 2001
From: Gao Pan <pandy.gao@nxp.com>
Date: Fri, 19 Jan 2018 11:12:15 +0800
Subject: [PATCH 3215/5242] MLK-17416 imx8: sim: add usleep_range() before
 reading SPDP Bit

commit  2f83fb22160406a4db7101f6d0f26bdcb88afb6c from
https://source.codeaurora.org/external/imx/linux-imx.git

Card Presence Detect Status Bit SPDP in EMV_SIM_PCSR is
synchronized by two posedge of low_ref_clk which is 32KHz.

So there should be 1.5 low_ref_clk cycles(about 90us) before
reading SPDP Bit.

Signed-off-by: Gao Pan <pandy.gao@nxp.com>
Acked-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/sim/imx_emvsim.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mxc/sim/imx_emvsim.c b/drivers/mxc/sim/imx_emvsim.c
index 164cf92..7747115 100644
--- a/drivers/mxc/sim/imx_emvsim.c
+++ b/drivers/mxc/sim/imx_emvsim.c
@@ -751,6 +751,7 @@ static void emvsim_start(struct emvsim_t *emvsim)
 	clk_div = (clk_rate + emvsim->clk_rate - 1) / emvsim->clk_rate;
 	__raw_writel(clk_div, emvsim->ioaddr + EMV_SIM_CLKCFG);
 
+	usleep_range(90, 100);
 	/* SPDP=0: SIM Presence Detect pin is low, default PRESENT status */
 	if (__raw_readl(emvsim->ioaddr + EMV_SIM_PCSR) & SPDP) {
 		emvsim->present = SIM_PRESENT_REMOVED;
-- 
1.7.9.5

